controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -284,6 +319,61 @@ func (r *StorageConsumerReconciler) reconcileCephBlockPool() error {
 	return nil
 }
 
+func (r *StorageConsumerReconciler) reconcileCephFilesystemSubVolumeGroup() error {
+
+	cephFilesystemList := rookCephv1.CephFilesystemList{}
+	if err := r.Client.List(r.ctx, &cephFilesystemList, client.InNamespace(r.namespace)); err != nil {
+		return fmt.Errorf("Error fetching CephFilesystemList. %+v", err)
+	}
+
+	var cephFileSystemName string
+	availableCephFileSystems := len(cephFilesystemList.Items)
+	if availableCephFileSystems == 0 {
+		return fmt.Errorf("No CephFileSystem found in the cluster")
+	} else if availableCephFileSystems > 1 {
+		klog.Warningf("More than one CephFileSystem found in the cluster, selecting the first one")
+	}
+
+	cephFileSystemName = cephFilesystemList.Items[0].Name
+
+	desired := &rookCephv1.CephFilesystemSubVolumeGroup{
+		Spec: rookCephv1.CephFilesystemSubVolumeGroupSpec{
+			FilesystemName: cephFileSystemName,
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephFilesystemSubVolumeGroup, func() error {
+		if err := r.own(r.cephFilesystemSubVolumeGroup); err != nil {
+			return err
+		}
+
+		r.cephFilesystemSubVolumeGroup.Spec = desired.Spec
+		return nil
+	})
```

Comment:
Why create `rookCephv1.CephFilesystemSubVolumeGroup` object if we are only using the spec and also why outside of the CreateOrUpdate call? 


```suggestion
	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephFilesystemSubVolumeGroup, func() error {
		if err := r.own(r.cephFilesystemSubVolumeGroup); err != nil {
			return err
		}

		r.cephFilesystemSubVolumeGroup.Spec = rookCephv1.CephFilesystemSubVolumeGroupSpec{
			FilesystemName: cephFileSystemName,
		}
		return nil
	})
```